cmd/goreleaser/internal/configure.go

```diff
@@ -90,19 +98,58 @@ func Build(dist string) config.Build {
 	}
 }
 
-func Archives(dist string) (r []config.Archive) {
+func BuildCGO(dist string) []config.Build {
+	return []config.Build{
+		{
+			ID:     dist,
+			Dir:    "_build",
+			Binary: dist,
+			BuildDetails: config.BuildDetails{
+				Env:     []string{"CGO_ENABLED=0"},
+				Flags:   []string{"-trimpath"},
+				Ldflags: []string{"-s", "-w"},
+			},
+			Goos:   []string{"linux", "windows"},
+			Goarch: Architectures,
+			Goarm:  ArmVersions,
+			Ignore: []config.IgnoredBuild{
+				{Goos: "windows", Goarch: "arm"},
+				{Goos: "windows", Goarch: "arm64"},
+				{Goos: "windows", Goarch: "s390x"},
+			},
+		},
+		{
+			ID:     getCgoDistId(dist),
+			Dir:    "_build",
+			Binary: dist,
+			BuildDetails: config.BuildDetails{
+				Env:     []string{"CGO_ENABLED=1", "CGO_LDFLAGS='-fstack-protector'"},
+				Flags:   []string{"-trimpath", "-buildmode=pie"},
+				Ldflags: []string{"-s", "-w"},
+			},
+			Goos:   []string{"darwin"},
+			Goarch: []string{"amd64", "arm64"},
+		},
+	}
+}
+
+func Archives(dist string, cgo string) (r []config.Archive) {
 	return []config.Archive{
-		Archive(dist),
+		Archive(dist, cgo),
 	}
 }
 
 // Archive configures a goreleaser archive (tarball).
 // https://goreleaser.com/customization/archive/
-func Archive(dist string) config.Archive {
+func Archive(dist string, cgo string) config.Archive {
+	builds := []string{dist}
+	if len(cgo) > 0 {
```

Comment:
what's the benefit of having cgo set as a string instead of a bool? Do we plan to have more conditional logic based on the content of `cgo` if we have more distributions built with cgo in the future?

Reply:
I put it to string in case we need other OS to support cgo in the future. I also wrote it in the cgoFlag description.